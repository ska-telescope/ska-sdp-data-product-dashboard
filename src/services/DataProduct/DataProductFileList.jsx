import axios from 'axios';
import mockFilesTree from '../Mocking/mockFilesTree';

async function DataProductFileList() {
  async function fetchFileList() {
    const apiUrl = process.env.SKA_SDP_DATA_PRODUCT_API_URL;
    const apiPort = process.env.SKA_SDP_DATA_PRODUCT_API_PORT;
    try {
      return await axios.get(`${apiUrl}:${apiPort}/filelist`, {
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json'
        }
      });
    } catch (e) {
      // eslint-disable-next-line no-console
      // console.error('Catch error', e);
      const noData = {
        id: 'root',
        name: 'SDP Data API not available',
        relativefilename: '',
        type: 'directory'
      };
      return noData;
    }
  }
  return JSON.parse(process.env.SKA_SDP_DATA_PRODUCT_DUMMY_DATA) ? mockFilesTree : fetchFileList();
}

export default DataProductFileList;
