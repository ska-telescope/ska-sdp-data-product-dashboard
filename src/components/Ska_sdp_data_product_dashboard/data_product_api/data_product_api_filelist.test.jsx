import { cleanup } from '@testing-library/react';
import DataProductFileList from './data_product_api_filelist';

process.env.REACT_APP_SKA_SDP_DATA_PRODUCT_DUMMY_DATA = 'true';

// afterEach function runs after each test suite is executed
afterEach(() => {
  cleanup(); // Resets the DOM after each test suite
});

it('renders user data', async () => {
  const originalValue = process.env.REACT_APP_SKA_SDP_DATA_PRODUCT_DUMMY_DATA;
  process.env.REACT_APP_SKA_SDP_DATA_PRODUCT_DUMMY_DATA = 'true';
  const fileList = await DataProductFileList();
  process.env.REACT_APP_SKA_SDP_DATA_PRODUCT_DUMMY_DATA = originalValue;

  expect(fileList).toMatchSnapshot();
});
