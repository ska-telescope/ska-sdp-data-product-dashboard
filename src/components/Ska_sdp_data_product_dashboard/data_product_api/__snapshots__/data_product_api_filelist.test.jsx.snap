// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data_product_api_filelist Retrieves user data 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "id": 2,
          "name": "Mock tree file 1.txt",
          "relativefilename": "testfile1.txt",
          "type": "file",
        },
        Object {
          "id": 3,
          "name": "Mock tree file 2.txt",
          "relativefilename": "testfile2.txt",
          "type": "file",
        },
      ],
      "id": 1,
      "name": "Mock tree folder",
      "relativefilename": "testfile1.txt",
      "type": "file",
    },
  ],
  "id": "root",
  "name": "SDP Data API not available",
  "relativefilename": ".",
  "type": "directory",
}
`;
